frontend/lib/verification/verification_view.dart

```diff
@@ -0,0 +1,226 @@
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
+
+import '../identification/base_card_details.dart';
+import '../qr_code_scanner/qr_code_scanner.dart';
+import 'remote_verification.dart';
+import 'verification_card_details_model.dart';
+import 'verification_error.dart';
+import 'verification_qr_content_parser.dart';
+
+class VerificationView extends StatefulWidget {
+  const VerificationView({
+    Key key,
+  }) : super(key: key);
+
+  @override
+  State<VerificationView> createState() => _VerificationViewState();
+}
+
+class _VerificationViewState extends State<VerificationView> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          leading: IconButton(
+            icon: Icon(
+              Icons.arrow_back_ios,
+              color: Theme.of(context).textTheme.bodyText1.color,
+            ),
+            onPressed: () => Navigator.of(context).pop(),
+          ),
+          title: Text("Karte verifizieren"),
+        ),
+        body: SingleChildScrollView(
+          child: Column(
+            children: <Widget>[
+              _buildContentCard(child: RichText(text: _buildInfoText())),
+              Center(
+                child: RaisedButton(
+                    padding: EdgeInsets.symmetric(vertical: 12, horizontal: 24),
+                    onPressed: _onScanCodePressed,
+                    color: Theme.of(context).primaryColor,
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(32.0),
+                    ),
+                    child: Text(
+                      "Code einscannen",
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline6
+                          .merge(TextStyle(color: Colors.white, fontSize: 20)),
+                    )),
+              ),
+              Consumer<VerificationCardDetailsModel>(
+                  builder: (context, verCardDetailsModel, child) {
+                return _buildVerificationResult(verCardDetailsModel);
+              }),
+            ],
+          ),
+        ));
+  }
+
+  Widget _buildVerificationResult(VerificationCardDetailsModel model) {
+    switch (model.verificationState) {
+      case VerificationState.waitingForScan:
+        return Container(width: 0.0, height: 0.0);
+      case VerificationState.verificationInProgress:
+        verifyCardValidWithRemote(model);
+        return _buildWaitingForVerificationResult();
+      case VerificationState.verificationSuccess:
+        return _buildPositiveVerificationResult(
+            model.verificationCardDetails.cardDetails);
+      case VerificationState.verificationFailure:
+        return _buildNegativeVerificationResult(model.verificationError);
+    }
+    return Container(width: 0.0, height: 0.0);
```

Comment:
I would rather return `null` here and in the first case, and just insert nothing in the tree when rendering.

Reply:
Returning null instead of a Widget causes an error

Reply:
Yes, because you can not insert null into the rendering tree, but I think we could just not insert it in case it is null

Reply:
Yeah, do you know how that works with the consumer above? Tbh i dont really care about that right now, it works and the verification page is very simple anyway

Reply:
Ah, sorry, I missed the Consumer thing. Yeah, leave it like that then 👍🏽